chain/src/main/scala/org/bitcoins/chain/blockchain/Blockchain.scala

```diff
@@ -37,7 +36,22 @@ case class Blockchain(headers: Vector[BlockHeaderDb])
   override def length: Int = headers.length
 
   /** @inheritdoc */
-  override def apply(idx: Int): BlockHeaderDb = headers(idx)
+  override def apply(idx: Int): BlockHeaderDb = headers.apply(idx)
+
+  /** Finds a block header at a given height */
+  def findAtHeight(height: Int): Option[BlockHeaderDb] =
+    find(_.height == height)
+
+  /** Splits the blockchain at the header, returning a new blockchain where the best tip is the given header */
+  def fromHeader(header: BlockHeaderDb): Option[Blockchain] = {
+    val headerIdxOpt = headers.zipWithIndex.find(_._1 == header)
```

Comment:
wouldn't it be more efficient to not `zipWithIndex` and instead to do something like `indexOf` after the fact (to avoid doing processing on the entire header chain)